services/mac_manager/src/test/java/com/futurewei/alcor/macmanager/controller/MacControllerTest.java

```diff
@@ -0,0 +1,111 @@
+*/
+package com.futurewei.alcor.macmanager.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.futurewei.alcor.macmanager.entity.MacState;
+import com.futurewei.alcor.macmanager.entity.MacStateJson;
+import com.futurewei.alcor.macmanager.service.MacAddressService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class MacControllerTest {
+
+    private static final ObjectMapper om = new ObjectMapper();
+
+    public MacState testMacState;
+    @Autowired
+    MacAddressService service;
+    String strTestMac = "";
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private MacController mockController;
+
+    @Before
+    public void init() {
+        MacState macState = new MacState("", "project1", "vpc1", "port1", "active");
+        MacStateJson macStateJson = new MacStateJson(macState);
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        try {
+            String json = objectMapper.writeValueAsString(macStateJson);
+            MacState macState2 = service.createMacState(macState);
+            strTestMac = macState2.getMacAddress();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void test_index() throws Exception {
+        this.mockMvc.perform(get("/start.html"))
+                .andDo(print())
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void test_createMacState() throws Exception {
+        MacState macState = new MacState("", "project1", "vpc1", "port2", "active");
+        MacStateJson macStateJson = new MacStateJson(macState);
+        ObjectMapper objectMapper = new ObjectMapper();
+        String json = objectMapper.writeValueAsString(macStateJson);
+
+        System.out.println(json);
+        MvcResult mvcResult = this.mockMvc.perform(post("/macs")
+                .content(json)
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andDo(print())
+                .andReturn();
```

Comment:
Do we need to verify the post return?

Reply:
Yes, this returns a created mac address, which is necessary information to test other APIs such as delete or get. Usually return is not needed, but in mac manager, we don't know Mac address until it is created.